### PR TITLE
Add playback rate control

### PR DESCRIPTION
**Pull request by da-moo**
Hey there! This is related to #44 where I asked about implementing this feature. This pull request is my first pass at implementing the control. Here are the details:

- I tried to reuse as much as I could via GB's own CSS classes to get the styling to match the existing player controls
  - Where I couldn't realistically reuse classes, I implemented my own and matched styling to maintain the same look and feel
- The control is an SVG icon on the right-hand side, prepended before the first icon (the share button)
  - I used [feather icons](/feathericons/feather)'s clock icon, but it doesn't look the nicest. I need some help adjusting it since I don't have a ton of experience with SVGs
- Clicking on the icon brings up a tooltip menu, implemented exactly how GB implements their video quality control
- The tooltip menu contains a slider that defaults to 1.00x playback speed, but is a range from 0.1x to 2.00x. 
- Adjusting playback rate works for both HTML5 & progressive video types (at least in Firefox, I haven't tested Chrome)

And that's about it. Let me know what you think, and if you have any tips on how to get the SVG looking nicer.

Screenshots:

firefox_2019-04-16_12-23-43

firefox_2019-04-16_12-23-59

### CODE REVIEW

**Comment by kmausolf** [comment]
Thank you so much for contributing!  One question:  Does this exclude live streams (GBI and live, of course)?

**Comment by da-moo** [comment]
> Does this exclude live streams (GBI and live, of course)?

Yup! In `manifest.json` I only load the new script where the URL matches the hompage or a show (exactly the same as the script for hiding *This is the Run* spoilers). When I watched live streams this past week the new icon didn't show.

File: add_playback_rate.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      .addClass('av-ctrl qol-playback-rate-button')
+      .append($('<i>').append(playbackRateIconFragment()))
+  );
+  playbackDiv.append(playbackRateMenuFragment());
+  return playbackDiv;
+}
+
+function playbackRateIconFragment() {
+  return `
+    <svg class="symbol" viewBox = "0 0 20 20" width="80" height="80">
```
Comment (kmausolf) [inline add_playback_rate.js]: Nice use of SVG here.

**Comment by lrkissling** [comment]
Whoops, I didn't see this until now! I'm a little less attentive to this project when the weather gets nice...

This seems super rad, I'll check it out and get back to you shortly!

File: add_playback_rate.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (da-moo) [reply add_playback_rate.js]: Implemented this change in 892b103, thanks!

File: giant_bomb_qol.css [hunk 2]
```diff
@@ -1,3 +1,42 @@
+/*************************************************
+Playback Rate Control
+*************************************************/
+
```
[the remark this answers is not in the document]
Comment (da-moo) [reply giant_bomb_qol.css]: Implemented this change in 892b103, thanks!

**Comment by da-moo** [comment]
While testing the new style changes, I was curious to see if this PR also happened to work for the podcast player on the site. Unfortunately, it doesn't work and there seems to be some issues with the control displaying on top of the podcast player.

image

I spent a little bit of time seeing if I could control the playback rate of the podcast player similar to how I do it for HTML5 videos, but it appears that there's no HTML5 `<audio>` element on the page to control. I'm guessing the GB engineers use some custom JavaScript to control playback. I could try asking the engineers what they use to play audio and if it's easy to find and manipulate, but in the mean time before this is merged I should find a way to disable the playback rate control from showing on the podcast player.

**Comment by lrkissling** [comment]
Good catch @da-moo I'm fine with disabling it for podcasts for now if you can't find a way of getting it working for them.

I also noticed that we will need this script to load for pages that being with `*://www.giantbomb.com/shows/*`. For example, [https://www.giantbomb.com/videos/dreaming-big-with-media-molecule/2300-14163/](https://www.giantbomb.com/videos/dreaming-big-with-media-molecule/2300-14163/)

**Comment by kmausolf** [comment]
I think it's fair to break this feature out into two separate pieces, as they both interface with different technology.  I agree with disabling this for podcasts for now.  We can workshop the podcast control feature in a different PR.

**Comment by lrkissling** [comment]
@da-moo Thanks so much for your work on this! It's available in the version 3 release of the extension!

**Pull request merged**